ex4/ex4.js

```diff
@@ -6,3 +6,32 @@
  * toNumber("71") === 71
  * toNumber("a7") === null
  */
+function toNumber(str) {
+
+	let tmpStr = '';
+
+	for (let i = 0; i < str.length; i++) {
+
+		if (str.slice(i,i+1) >= 0 && str.slice(i,i+1) <= 9) {
+
+			tmpStr += str.slice(i,i+1);  
+
+		} else {
+
+			return null;
+
+		}
+	}
+
+	return Number(tmpStr);
+}
+
+function testFn(str) {
+
+	console.assert(toNumber(str) !== null, toNumber(str));
+}
+
+testFn("71");
```

Comment:
Once you fix the tests, you'll see the code is not doing what it's supposed to be doing.